src/components/ClusterNodes/types.ts

```diff
@@ -0,0 +1,214 @@
+import { ResponseType } from '../../services/service.types'
+import { LabelTag } from '../app/types'
+
+export interface ResourceDetail {
+    name: string
+    capacity: string
+    allocatable: string
+    usage: string
+    request: string
+    limit: string
+    usagePercentage: string
+    requestPercentage: string
+    limitPercentage: string
+}
+export interface ClusterCapacityType {
+    nodeK8sVersions: string[]
+    nodeErrors: Record<string, string>[]
+    cpu: ResourceDetail
+    memory: ResourceDetail
+}
+export interface ClusterDetail {
+    id: number
+    name: string
+    nodeCount: number
+    nodeErrors: Record<string, string>[]
+    errorInNodeListing: string
+    nodeK8sVersions: string[]
+    cpu: ResourceDetail
+    memory: ResourceDetail
+}
+
+export interface NodeRowDetail {
+    name: string
+    status: string
+    roles: string[]
+    errors: Record<string, string>[]
+    k8sVersion: string
+    podCount: number
+    taintCount: number
+    cpu: ResourceDetail
+    memory: ResourceDetail
+    age: string
+}
+
+export interface ClusterListResponse extends ResponseType {
+    result?: ClusterDetail[]
+}
+export interface ClusterCapacityResponse extends ResponseType {
+    result?: ClusterCapacityType
+}
+export interface NodeListResponse extends ResponseType {
+    result?: NodeRowDetail[]
+}
+
+export interface PodType {
+    name: string
+    namespace: string
+    cpu: ResourceDetail
+    memory: ResourceDetail
+    age: string
+}
+export interface NodeDetail {
+    name: string
+    clusterName: string
+    status: string
+    version: string
+    kind: string
+    roles: string[]
+    k8sVersion: string
+    errors: Record<string, string>
+    internalIp: string
+    externalIp: string
+    unschedulable: boolean
+    createdAt: string
+    labels: LabelTag[]
+    annotations: LabelTag[]
+    taints: LabelTag[]
+    resources: ResourceDetail[]
+    pods: PodType[]
+    manifest: object
+    conditions: { haveIssue: boolean; message: string; reason: string; type: string }[]
+    taintCount: number
+}
+export interface NodeDetailResponse extends ResponseType {
+    result?: NodeDetail
+}
+
+export interface UpdateNodeRequestBody {
+    clusterId: number
+    name: string
+    manifestPatch: string
+    version: string
+    kind: string
+}
+
+export interface columnMetadataType {
+    sortType: string
+    columnIndex: number
+    label: string
+    value: string
+    isDefault?: boolean
+    isSortingAllowed?: boolean
+    sortingFieldName?: string
+    isDisabled?: boolean
+}
+
+export const TEXT_COLOR_CLASS = {
+    Ready: 'cg-5',
+    'Not ready': 'cr-5',
+}
+
+export const columnMetadata: columnMetadataType[] = [
```

Comment:
Suggestion - Since it's a const we can name it something like - `COLUMN_METADATA`.